extension/src/SymbolReference/SymbolReferenceCache.ts

```diff
@@ -11,30 +11,28 @@ export class SymbolReferenceCache {
     this.cache = new Map();
   }
 
-  private id(app: AppIdentifier): string {
-    return `${app.name}-${app.publisher}-${app.version}`;
+  static key(appId: AppIdentifier): string {
+    return `${appId.name}-${appId.publisher}-${appId.version}`;
   }
 
-  get(app: AppIdentifier): AppPackage | undefined {
-    return this.cache.get(this.id(app));
+  get(appId: AppIdentifier): AppPackage | undefined {
+    return this.cache.get(SymbolReferenceCache.key(appId));
   }
 
   set(appPackage: AppPackage): void {
     if (this.isCached(appPackage)) {
       return;
     }
     const appToCache = appPackage;
-    //TODO: Test if objects share reference
-    appToCache.symbolReference = undefined; // Free up unnecessary memory allocation
-    this.cache.set(this.id(appToCache), appToCache);
+    this.cache.set(SymbolReferenceCache.key(appToCache), appToCache);
   }
 
-  isCached(app: AppIdentifier): boolean {
-    return this.cache.get(this.id(app)) !== undefined;
+  isCached(appId: AppIdentifier): boolean {
+    return this.cache.get(SymbolReferenceCache.key(appId)) !== undefined;
```

Comment:
this cache can keep quite large objects, so using `has()` instead of `get()` should have advantages, shouldn't it?

Reply:
I would assume so. Fixed :+1: